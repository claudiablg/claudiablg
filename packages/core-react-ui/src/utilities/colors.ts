import { Color } from '@newrade/core-design-system';

export function getHSLColor({ h, s, l, a }: Partial<Color> = { h: 200, s: 100, l: 100, a: 1 }) {
  return `hsla(${h}, ${s}%, ${l}%, ${a || 1})`;
}

export function lightenColor(color: Color, amount: number) {}

export function darkenColor(color: Color, amount: number) {}
