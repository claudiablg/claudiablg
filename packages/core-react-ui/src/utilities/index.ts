export * from './colors.utilities';
export * from './component.utilities';
export * from './css-variable.utilities';
export * from './effects.utilities';
export * from './filter-color.utility';
export * from './text.utilities';
export * from './theme.utilities';
export * from './utilities';
