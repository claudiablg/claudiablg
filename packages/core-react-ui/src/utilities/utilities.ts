import { ColorIntents, Colors } from '@newrade/core-design-system';

/**
 * Create default color intents.
 */
export function createDefaultColorIntents(colors: Colors): ColorIntents {
  return {
    primary: colors.primary500,
    primaryReversed: colors.greyscale0,
    secondary: colors.primary500,
    secondaryReversed: colors.greyscale0,
    primaryText: colors.greyscale900,
    primaryTextReversed: colors.greyscale0,
    secondaryText: colors.greyscale700,
    secondaryTextReversed: colors.greyscale0,
    tertiaryText: colors.greyscale600,
    tertiaryTextReversed: colors.greyscale0,
    successText: colors.utilityGreen1000,
    successAction: colors.utilityGreen500,
    successBackground: colors.utilityGreen100,
    warningText: colors.utilityYellow1000,
    warningAction: colors.utilityYellow500,
    warningBackground: colors.utilityYellow100,
    dangerText: colors.utilityRed1000,
    dangerAction: colors.utilityRed500,
    dangerBackground: colors.utilityRed100,
    background0: colors.greyscale0,
    background1: colors.greyscale100,
    background2: colors.greyscale200,
  };
}

/**
 * Convert a number (in px) into a rem value.
 * @param px value in pixel
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxToRem({ baseUnitPx, value }: { baseUnitPx: number; value: number }): string {
  return `${value / baseUnitPx}rem`;
}

/**
 * Convert a number (in px) into a em relative value.
 * @param px value in pixel
 * @param fontSize font-size set on the element
 */
export function pxToEm({ fontSize, value }: { fontSize: number; value: number }): string {
  return `${value / fontSize}em`;
}

/**
 * Convert px value (e.g. '15px') into a rem value.
 * @param px value in string px value
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxStringToRem({ baseFontSize, value }: { baseFontSize: number; value: string }): string {
  if (!value?.length) {
    return '';
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return '';
  }
  const conversion = Number(match[0]);
  if (!conversion) {
    return '';
  }

  return `${conversion / baseFontSize}rem`;
}

/**
 * Convert px value (e.g. '15px') into a number value.
 * @param px value in string px value
 */
export function pxStringToNumber({ value }: { value: string }): number | undefined {
  if (!value?.length) {
    return undefined;
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return undefined;
  }
  const conversion = Number(match[0]);
  if (conversion === undefined || conversion === NaN || conversion === null) {
    return undefined;
  }

  return conversion;
}
