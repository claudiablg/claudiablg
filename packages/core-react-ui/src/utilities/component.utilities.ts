import { pascal } from 'case';
import { keys } from './utilities';

/**
 * <Link /> => "Link Medium Underline"
 */
export function getDefaultTextFromProps(name: string, props: object): string {
  const propString = `${keys(props)
    .map((prop) => pascal(props[prop]))
    .join(' ')}`;

  return `${pascal(name)} ${propString}`;
}
