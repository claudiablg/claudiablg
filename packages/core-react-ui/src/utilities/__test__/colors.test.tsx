import React from 'react';
import { Color } from '@newrade/core-design-system';
import { cleanup, render } from '@testing-library/react';
import { getCSSColor, getCSSVarNamesForColors, getCSSVarForColors, getCSSColors } from '../colors.utilities';
import { defaultColorsColors, defaultColorIntents, defaultColors } from '../../design-system/default-colors';

describe('colors utilities', () => {
  describe(`${getCSSColor.name}`, () => {
    it('should create a valid CSS color from a Color object', () => {
      const color: Color = {
        h: 222,
        s: 40,
        l: 50,
        a: 80,
      };
      const cssColor = getCSSColor(color);
      expect(cssColor).toBe('hsl(222deg 40% 50% / 80%)');
    });

    it('should create a valid CSS color from a Color object with no transparency', () => {
      const color: Color = {
        h: 222,
        s: 40,
        l: 50,
        a: 100,
      };
      const cssColor = getCSSColor(color);
      expect(cssColor).toBe('hsl(222deg 40% 50% / 100%)');
    });

    it(`should create a valid CSS color from a Color object the param 'a' not set`, () => {
      const color: Color = {
        h: 222,
        s: 40,
        l: 50,
      };
      const cssColor = getCSSColor(color);
      expect(cssColor).toBe('hsl(222deg 40% 50% / 100%)');
    });

    it(`should create a valid CSS color that can be applied to a div`, () => {
      const color: Color = {
        h: 222,
        s: 40,
        l: 50,
      };
      const result = render(<div className="1" style={{ backgroundColor: getCSSColor(color) }}></div>);
      expect(result.container.firstChild).toBeDefined();
      if (result.container.firstChild) {
        const element = result.container.getElementsByClassName('1');
        if (!element) {
          fail('element div not found');
        }
        // @ts-ignore
        expect(element[0]).toHaveStyle(`backgroundColor: hsl(222deg 40% 50% / 100%)`);
      }
      cleanup();
    });
  });

  describe(`${getCSSVarNamesForColors.name}`, () => {
    it(`should generate CSS variable names for a given Colors object`, () => {
      const vars = getCSSVarNamesForColors({ colors: defaultColorsColors, colorIntents: defaultColorIntents });
      expect(vars).toMatchSnapshot();
    });

    it(`should generate CSS statement to access Colors CSS vars`, () => {
      const vars = getCSSVarForColors({ colors: defaultColorsColors, colorIntents: defaultColorIntents });
      expect(vars).toMatchSnapshot();
    });
  });

  describe(`${getCSSColors.name}`, () => {
    it(`should generate a CSS colors from a Colors object`, () => {
      const colors = getCSSColors(defaultColors);
      expect(colors).toMatchSnapshot();
    });
  });
});
