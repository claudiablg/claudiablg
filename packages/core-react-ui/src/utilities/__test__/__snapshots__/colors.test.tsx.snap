// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors utilities getCSSColors should generate a CSS colors from a Colors object 1`] = `
Object {
  "colorIntents": Object {
    "accessibilityColor": "hsl(222deg 50% 36% / 100%)",
    "background0": "hsl(10deg 0% 100% / 100%)",
    "background1": "hsl(10deg 0% 97% / 100%)",
    "background2": "hsl(10deg 3% 93% / 100%)",
    "backgroundDisabled": "hsl(10deg 6% 90% / 100%)",
    "current": "currentColor",
    "dangerAction": "hsl(360deg 84% 48% / 100%)",
    "dangerBackground": "hsl(360deg 84% 82% / 100%)",
    "dangerText": "hsl(360deg 84% 14% / 100%)",
    "disabledText": "hsl(10deg 6% 73% / 100%)",
    "disabledTextReversed": "hsl(10deg 6% 90% / 100%)",
    "primary": "hsl(222deg 50% 50% / 100%)",
    "primaryReversed": "hsl(10deg 10% 100% / 100%)",
    "primaryText": "hsl(10deg 6% 7% / 100%)",
    "primaryTextReversed": "hsl(10deg 10% 100% / 100%)",
    "secondary": "hsl(200deg 50% 50% / 100%)",
    "secondaryReversed": "hsl(10deg 10% 100% / 100%)",
    "secondaryText": "hsl(10deg 5% 46% / 100%)",
    "secondaryTextReversed": "hsl(10deg 10% 100% / 100%)",
    "successAction": "hsl(112deg 71% 53% / 100%)",
    "successBackground": "hsl(112deg 71% 87% / 100%)",
    "successText": "hsl(112deg 71% 19% / 100%)",
    "tertiaryText": "hsl(10deg 5% 66% / 100%)",
    "tertiaryTextReversed": "hsl(10deg 10% 100% / 100%)",
    "transparent": "transparent",
    "warningAction": "hsl(47deg 90% 54% / 100%)",
    "warningBackground": "hsl(47deg 90% 88% / 100%)",
    "warningText": "hsl(47deg 90% 20% / 100%)",
  },
  "colors": Object {
    "accent1": Object {
      "100": "hsl(200deg 50% 78% / 100%)",
      "300": "hsl(200deg 50% 64% / 100%)",
      "500": "hsl(200deg 50% 50% / 100%)",
      "700": "hsl(200deg 50% 36% / 100%)",
      "900": "hsl(200deg 50% 22% / 100%)",
      "baseHue": 200,
      "baseSat": 50,
    },
    "accent2": Object {
      "100": "hsl(200deg 50% 78% / 100%)",
      "300": "hsl(200deg 50% 64% / 100%)",
      "500": "hsl(200deg 50% 50% / 100%)",
      "700": "hsl(200deg 50% 36% / 100%)",
      "900": "hsl(200deg 50% 22% / 100%)",
      "baseHue": 200,
      "baseSat": 50,
    },
    "accent3": Object {
      "100": "hsl(200deg 50% 78% / 100%)",
      "300": "hsl(200deg 50% 64% / 100%)",
      "500": "hsl(200deg 50% 50% / 100%)",
      "700": "hsl(200deg 50% 36% / 100%)",
      "900": "hsl(200deg 50% 22% / 100%)",
      "baseHue": 200,
      "baseSat": 50,
    },
    "current": "currentColor",
    "effectTransparentHeavy": "hsl(0deg 0% 0% / 13%)",
    "effectTransparentLight": "hsl(0deg 0% 0% / 10%)",
    "effectTransparentMedium": "hsl(0deg 0% 0% / 10%)",
    "grey": Object {
      "0": "hsl(10deg 0% 100% / 100%)",
      "0-reversed": "hsl(10deg 10% 100% / 100%)",
      "100": "hsl(10deg 6% 90% / 100%)",
      "100-reversed": "hsl(10deg 10% 0% / 100%)",
      "1000": "hsl(10deg 6% 7% / 100%)",
      "200": "hsl(10deg 6% 86% / 100%)",
      "200-reversed": "hsl(10deg 10% 0% / 100%)",
      "25": "hsl(10deg 0% 97% / 100%)",
      "300": "hsl(10deg 6% 83% / 100%)",
      "400": "hsl(10deg 6% 73% / 100%)",
      "50": "hsl(10deg 3% 93% / 100%)",
      "500": "hsl(10deg 5% 66% / 100%)",
      "600": "hsl(10deg 5% 59% / 100%)",
      "700": "hsl(10deg 5% 46% / 100%)",
      "800": "hsl(10deg 4% 26% / 100%)",
      "900": "hsl(10deg 4% 13% / 100%)",
      "baseHue": 10,
      "baseSat": 6,
    },
    "primary": Object {
      "100": "hsl(222deg 50% 78% / 100%)",
      "300": "hsl(222deg 50% 64% / 100%)",
      "500": "hsl(222deg 50% 50% / 100%)",
      "700": "hsl(222deg 50% 36% / 100%)",
      "900": "hsl(222deg 50% 22% / 100%)",
      "baseHue": 222,
      "baseSat": 50,
    },
    "transparent": "hsl(0deg 0% 0% / 0%)",
    "utilityGreen": Object {
      "100": "hsl(112deg 71% 87% / 100%)",
      "300": "hsl(112deg 71% 70% / 100%)",
      "500": "hsl(112deg 71% 53% / 100%)",
      "700": "hsl(112deg 71% 36% / 100%)",
      "900": "hsl(112deg 71% 19% / 100%)",
      "baseHue": 112,
      "baseSat": 71,
    },
    "utilityRed": Object {
      "100": "hsl(360deg 84% 82% / 100%)",
      "300": "hsl(360deg 84% 65% / 100%)",
      "500": "hsl(360deg 84% 48% / 100%)",
      "700": "hsl(360deg 84% 31% / 100%)",
      "900": "hsl(360deg 84% 14% / 100%)",
      "baseHue": 360,
      "baseSat": 84,
    },
    "utilityYellow": Object {
      "100": "hsl(47deg 90% 88% / 100%)",
      "300": "hsl(47deg 90% 71% / 100%)",
      "500": "hsl(47deg 90% 54% / 100%)",
      "700": "hsl(47deg 90% 37% / 100%)",
      "900": "hsl(47deg 90% 20% / 100%)",
      "baseHue": 47,
      "baseSat": 90,
    },
  },
  "gradients": Object {
    "gradient1": "linear-gradient(45deg, hsl(200deg 50% 78% / 100%) ,hsl(200deg 50% 50% / 100%) 100)",
    "gradient2": "linear-gradient(45deg, hsl(200deg 50% 78% / 100%) ,hsl(200deg 50% 50% / 100%) 100)",
    "gradient3": "linear-gradient(45deg, hsl(200deg 50% 78% / 100%) ,hsl(200deg 50% 50% / 100%) 100)",
  },
  "var": Object {
    "colorIntents": Object {
      "accessibilityColor": "var(--color-intent-accessibility-color)",
      "background0": "var(--color-intent-background0)",
      "background1": "var(--color-intent-background1)",
      "background2": "var(--color-intent-background2)",
      "backgroundDisabled": "var(--color-intent-background-disabled)",
      "current": "var(--color-intent-current)",
      "dangerAction": "var(--color-intent-danger-action)",
      "dangerBackground": "var(--color-intent-danger-background)",
      "dangerText": "var(--color-intent-danger-text)",
      "disabledText": "var(--color-intent-disabled-text)",
      "disabledTextReversed": "var(--color-intent-disabled-text-reversed)",
      "primary": "var(--color-intent-primary)",
      "primaryReversed": "var(--color-intent-primary-reversed)",
      "primaryText": "var(--color-intent-primary-text)",
      "primaryTextReversed": "var(--color-intent-primary-text-reversed)",
      "secondary": "var(--color-intent-secondary)",
      "secondaryReversed": "var(--color-intent-secondary-reversed)",
      "secondaryText": "var(--color-intent-secondary-text)",
      "secondaryTextReversed": "var(--color-intent-secondary-text-reversed)",
      "successAction": "var(--color-intent-success-action)",
      "successBackground": "var(--color-intent-success-background)",
      "successText": "var(--color-intent-success-text)",
      "tertiaryText": "var(--color-intent-tertiary-text)",
      "tertiaryTextReversed": "var(--color-intent-tertiary-text-reversed)",
      "transparent": "var(--color-intent-transparent)",
      "warningAction": "var(--color-intent-warning-action)",
      "warningBackground": "var(--color-intent-warning-background)",
      "warningText": "var(--color-intent-warning-text)",
    },
    "colors": Object {
      "accent1": Object {
        "100": "var(--color-accent1-100)",
        "300": "var(--color-accent1-300)",
        "500": "var(--color-accent1-500)",
        "700": "var(--color-accent1-700)",
        "900": "var(--color-accent1-900)",
        "baseHue": "var(--color-accent1-base-hue)",
        "baseSat": "var(--color-accent1-base-sat)",
      },
      "accent2": Object {
        "100": "var(--color-accent2-100)",
        "300": "var(--color-accent2-300)",
        "500": "var(--color-accent2-500)",
        "700": "var(--color-accent2-700)",
        "900": "var(--color-accent2-900)",
        "baseHue": "var(--color-accent2-base-hue)",
        "baseSat": "var(--color-accent2-base-sat)",
      },
      "accent3": Object {
        "100": "var(--color-accent3-100)",
        "300": "var(--color-accent3-300)",
        "500": "var(--color-accent3-500)",
        "700": "var(--color-accent3-700)",
        "900": "var(--color-accent3-900)",
        "baseHue": "var(--color-accent3-base-hue)",
        "baseSat": "var(--color-accent3-base-sat)",
      },
      "current": "var(--color-current)",
      "effectTransparentHeavy": "var(--color-effect-transparent-heavy)",
      "effectTransparentLight": "var(--color-effect-transparent-light)",
      "effectTransparentMedium": "var(--color-effect-transparent-medium)",
      "grey": Object {
        "0": "var(--color-grey-0)",
        "0-reversed": "var(--color-grey-0-reversed)",
        "100": "var(--color-grey-100)",
        "100-reversed": "var(--color-grey-100-reversed)",
        "1000": "var(--color-grey-1000)",
        "200": "var(--color-grey-200)",
        "200-reversed": "var(--color-grey-200-reversed)",
        "25": "var(--color-grey-25)",
        "300": "var(--color-grey-300)",
        "400": "var(--color-grey-400)",
        "50": "var(--color-grey-50)",
        "500": "var(--color-grey-500)",
        "600": "var(--color-grey-600)",
        "700": "var(--color-grey-700)",
        "800": "var(--color-grey-800)",
        "900": "var(--color-grey-900)",
        "baseHue": "var(--color-grey-base-hue)",
        "baseSat": "var(--color-grey-base-sat)",
      },
      "primary": Object {
        "100": "var(--color-primary-100)",
        "300": "var(--color-primary-300)",
        "500": "var(--color-primary-500)",
        "700": "var(--color-primary-700)",
        "900": "var(--color-primary-900)",
        "baseHue": "var(--color-primary-base-hue)",
        "baseSat": "var(--color-primary-base-sat)",
      },
      "transparent": "var(--color-transparent)",
      "utilityGreen": Object {
        "100": "var(--color-utility-green-100)",
        "300": "var(--color-utility-green-300)",
        "500": "var(--color-utility-green-500)",
        "700": "var(--color-utility-green-700)",
        "900": "var(--color-utility-green-900)",
        "baseHue": "var(--color-utility-green-base-hue)",
        "baseSat": "var(--color-utility-green-base-sat)",
      },
      "utilityRed": Object {
        "100": "var(--color-utility-red-100)",
        "300": "var(--color-utility-red-300)",
        "500": "var(--color-utility-red-500)",
        "700": "var(--color-utility-red-700)",
        "900": "var(--color-utility-red-900)",
        "baseHue": "var(--color-utility-red-base-hue)",
        "baseSat": "var(--color-utility-red-base-sat)",
      },
      "utilityYellow": Object {
        "100": "var(--color-utility-yellow-100)",
        "300": "var(--color-utility-yellow-300)",
        "500": "var(--color-utility-yellow-500)",
        "700": "var(--color-utility-yellow-700)",
        "900": "var(--color-utility-yellow-900)",
        "baseHue": "var(--color-utility-yellow-base-hue)",
        "baseSat": "var(--color-utility-yellow-base-sat)",
      },
    },
  },
  "varNames": Object {
    "colorIntents": Object {
      "accessibilityColor": "--color-intent-accessibility-color",
      "background0": "--color-intent-background0",
      "background1": "--color-intent-background1",
      "background2": "--color-intent-background2",
      "backgroundDisabled": "--color-intent-background-disabled",
      "current": "--color-intent-current",
      "dangerAction": "--color-intent-danger-action",
      "dangerBackground": "--color-intent-danger-background",
      "dangerText": "--color-intent-danger-text",
      "disabledText": "--color-intent-disabled-text",
      "disabledTextReversed": "--color-intent-disabled-text-reversed",
      "primary": "--color-intent-primary",
      "primaryReversed": "--color-intent-primary-reversed",
      "primaryText": "--color-intent-primary-text",
      "primaryTextReversed": "--color-intent-primary-text-reversed",
      "secondary": "--color-intent-secondary",
      "secondaryReversed": "--color-intent-secondary-reversed",
      "secondaryText": "--color-intent-secondary-text",
      "secondaryTextReversed": "--color-intent-secondary-text-reversed",
      "successAction": "--color-intent-success-action",
      "successBackground": "--color-intent-success-background",
      "successText": "--color-intent-success-text",
      "tertiaryText": "--color-intent-tertiary-text",
      "tertiaryTextReversed": "--color-intent-tertiary-text-reversed",
      "transparent": "--color-intent-transparent",
      "warningAction": "--color-intent-warning-action",
      "warningBackground": "--color-intent-warning-background",
      "warningText": "--color-intent-warning-text",
    },
    "colors": Object {
      "accent1": Object {
        "100": "--color-accent1-100",
        "300": "--color-accent1-300",
        "500": "--color-accent1-500",
        "700": "--color-accent1-700",
        "900": "--color-accent1-900",
        "baseHue": "--color-accent1-base-hue",
        "baseSat": "--color-accent1-base-sat",
      },
      "accent2": Object {
        "100": "--color-accent2-100",
        "300": "--color-accent2-300",
        "500": "--color-accent2-500",
        "700": "--color-accent2-700",
        "900": "--color-accent2-900",
        "baseHue": "--color-accent2-base-hue",
        "baseSat": "--color-accent2-base-sat",
      },
      "accent3": Object {
        "100": "--color-accent3-100",
        "300": "--color-accent3-300",
        "500": "--color-accent3-500",
        "700": "--color-accent3-700",
        "900": "--color-accent3-900",
        "baseHue": "--color-accent3-base-hue",
        "baseSat": "--color-accent3-base-sat",
      },
      "current": "--color-current",
      "effectTransparentHeavy": "--color-effect-transparent-heavy",
      "effectTransparentLight": "--color-effect-transparent-light",
      "effectTransparentMedium": "--color-effect-transparent-medium",
      "grey": Object {
        "0": "--color-grey-0",
        "0-reversed": "--color-grey-0-reversed",
        "100": "--color-grey-100",
        "100-reversed": "--color-grey-100-reversed",
        "1000": "--color-grey-1000",
        "200": "--color-grey-200",
        "200-reversed": "--color-grey-200-reversed",
        "25": "--color-grey-25",
        "300": "--color-grey-300",
        "400": "--color-grey-400",
        "50": "--color-grey-50",
        "500": "--color-grey-500",
        "600": "--color-grey-600",
        "700": "--color-grey-700",
        "800": "--color-grey-800",
        "900": "--color-grey-900",
        "baseHue": "--color-grey-base-hue",
        "baseSat": "--color-grey-base-sat",
      },
      "primary": Object {
        "100": "--color-primary-100",
        "300": "--color-primary-300",
        "500": "--color-primary-500",
        "700": "--color-primary-700",
        "900": "--color-primary-900",
        "baseHue": "--color-primary-base-hue",
        "baseSat": "--color-primary-base-sat",
      },
      "transparent": "--color-transparent",
      "utilityGreen": Object {
        "100": "--color-utility-green-100",
        "300": "--color-utility-green-300",
        "500": "--color-utility-green-500",
        "700": "--color-utility-green-700",
        "900": "--color-utility-green-900",
        "baseHue": "--color-utility-green-base-hue",
        "baseSat": "--color-utility-green-base-sat",
      },
      "utilityRed": Object {
        "100": "--color-utility-red-100",
        "300": "--color-utility-red-300",
        "500": "--color-utility-red-500",
        "700": "--color-utility-red-700",
        "900": "--color-utility-red-900",
        "baseHue": "--color-utility-red-base-hue",
        "baseSat": "--color-utility-red-base-sat",
      },
      "utilityYellow": Object {
        "100": "--color-utility-yellow-100",
        "300": "--color-utility-yellow-300",
        "500": "--color-utility-yellow-500",
        "700": "--color-utility-yellow-700",
        "900": "--color-utility-yellow-900",
        "baseHue": "--color-utility-yellow-base-hue",
        "baseSat": "--color-utility-yellow-base-sat",
      },
    },
  },
}
`;

exports[`colors utilities getCSSVarNamesForColors should generate CSS statement to access Colors CSS vars 1`] = `
Object {
  "colorIntents": Object {
    "accessibilityColor": "var(--color-intent-accessibility-color)",
    "background0": "var(--color-intent-background0)",
    "background1": "var(--color-intent-background1)",
    "background2": "var(--color-intent-background2)",
    "backgroundDisabled": "var(--color-intent-background-disabled)",
    "current": "var(--color-intent-current)",
    "dangerAction": "var(--color-intent-danger-action)",
    "dangerBackground": "var(--color-intent-danger-background)",
    "dangerText": "var(--color-intent-danger-text)",
    "disabledText": "var(--color-intent-disabled-text)",
    "disabledTextReversed": "var(--color-intent-disabled-text-reversed)",
    "primary": "var(--color-intent-primary)",
    "primaryReversed": "var(--color-intent-primary-reversed)",
    "primaryText": "var(--color-intent-primary-text)",
    "primaryTextReversed": "var(--color-intent-primary-text-reversed)",
    "secondary": "var(--color-intent-secondary)",
    "secondaryReversed": "var(--color-intent-secondary-reversed)",
    "secondaryText": "var(--color-intent-secondary-text)",
    "secondaryTextReversed": "var(--color-intent-secondary-text-reversed)",
    "successAction": "var(--color-intent-success-action)",
    "successBackground": "var(--color-intent-success-background)",
    "successText": "var(--color-intent-success-text)",
    "tertiaryText": "var(--color-intent-tertiary-text)",
    "tertiaryTextReversed": "var(--color-intent-tertiary-text-reversed)",
    "transparent": "var(--color-intent-transparent)",
    "warningAction": "var(--color-intent-warning-action)",
    "warningBackground": "var(--color-intent-warning-background)",
    "warningText": "var(--color-intent-warning-text)",
  },
  "colors": Object {
    "accent1": Object {
      "100": "var(--color-accent1-100)",
      "300": "var(--color-accent1-300)",
      "500": "var(--color-accent1-500)",
      "700": "var(--color-accent1-700)",
      "900": "var(--color-accent1-900)",
      "baseHue": "var(--color-accent1-base-hue)",
      "baseSat": "var(--color-accent1-base-sat)",
    },
    "accent2": Object {
      "100": "var(--color-accent2-100)",
      "300": "var(--color-accent2-300)",
      "500": "var(--color-accent2-500)",
      "700": "var(--color-accent2-700)",
      "900": "var(--color-accent2-900)",
      "baseHue": "var(--color-accent2-base-hue)",
      "baseSat": "var(--color-accent2-base-sat)",
    },
    "accent3": Object {
      "100": "var(--color-accent3-100)",
      "300": "var(--color-accent3-300)",
      "500": "var(--color-accent3-500)",
      "700": "var(--color-accent3-700)",
      "900": "var(--color-accent3-900)",
      "baseHue": "var(--color-accent3-base-hue)",
      "baseSat": "var(--color-accent3-base-sat)",
    },
    "current": "var(--color-current)",
    "effectTransparentHeavy": "var(--color-effect-transparent-heavy)",
    "effectTransparentLight": "var(--color-effect-transparent-light)",
    "effectTransparentMedium": "var(--color-effect-transparent-medium)",
    "grey": Object {
      "0": "var(--color-grey-0)",
      "0-reversed": "var(--color-grey-0-reversed)",
      "100": "var(--color-grey-100)",
      "100-reversed": "var(--color-grey-100-reversed)",
      "1000": "var(--color-grey-1000)",
      "200": "var(--color-grey-200)",
      "200-reversed": "var(--color-grey-200-reversed)",
      "25": "var(--color-grey-25)",
      "300": "var(--color-grey-300)",
      "400": "var(--color-grey-400)",
      "50": "var(--color-grey-50)",
      "500": "var(--color-grey-500)",
      "600": "var(--color-grey-600)",
      "700": "var(--color-grey-700)",
      "800": "var(--color-grey-800)",
      "900": "var(--color-grey-900)",
      "baseHue": "var(--color-grey-base-hue)",
      "baseSat": "var(--color-grey-base-sat)",
    },
    "primary": Object {
      "100": "var(--color-primary-100)",
      "300": "var(--color-primary-300)",
      "500": "var(--color-primary-500)",
      "700": "var(--color-primary-700)",
      "900": "var(--color-primary-900)",
      "baseHue": "var(--color-primary-base-hue)",
      "baseSat": "var(--color-primary-base-sat)",
    },
    "transparent": "var(--color-transparent)",
    "utilityGreen": Object {
      "100": "var(--color-utility-green-100)",
      "300": "var(--color-utility-green-300)",
      "500": "var(--color-utility-green-500)",
      "700": "var(--color-utility-green-700)",
      "900": "var(--color-utility-green-900)",
      "baseHue": "var(--color-utility-green-base-hue)",
      "baseSat": "var(--color-utility-green-base-sat)",
    },
    "utilityRed": Object {
      "100": "var(--color-utility-red-100)",
      "300": "var(--color-utility-red-300)",
      "500": "var(--color-utility-red-500)",
      "700": "var(--color-utility-red-700)",
      "900": "var(--color-utility-red-900)",
      "baseHue": "var(--color-utility-red-base-hue)",
      "baseSat": "var(--color-utility-red-base-sat)",
    },
    "utilityYellow": Object {
      "100": "var(--color-utility-yellow-100)",
      "300": "var(--color-utility-yellow-300)",
      "500": "var(--color-utility-yellow-500)",
      "700": "var(--color-utility-yellow-700)",
      "900": "var(--color-utility-yellow-900)",
      "baseHue": "var(--color-utility-yellow-base-hue)",
      "baseSat": "var(--color-utility-yellow-base-sat)",
    },
  },
}
`;

exports[`colors utilities getCSSVarNamesForColors should generate CSS variable names for a given Colors object 1`] = `
Object {
  "colorIntents": Object {
    "accessibilityColor": "--color-intent-accessibility-color",
    "background0": "--color-intent-background0",
    "background1": "--color-intent-background1",
    "background2": "--color-intent-background2",
    "backgroundDisabled": "--color-intent-background-disabled",
    "current": "--color-intent-current",
    "dangerAction": "--color-intent-danger-action",
    "dangerBackground": "--color-intent-danger-background",
    "dangerText": "--color-intent-danger-text",
    "disabledText": "--color-intent-disabled-text",
    "disabledTextReversed": "--color-intent-disabled-text-reversed",
    "primary": "--color-intent-primary",
    "primaryReversed": "--color-intent-primary-reversed",
    "primaryText": "--color-intent-primary-text",
    "primaryTextReversed": "--color-intent-primary-text-reversed",
    "secondary": "--color-intent-secondary",
    "secondaryReversed": "--color-intent-secondary-reversed",
    "secondaryText": "--color-intent-secondary-text",
    "secondaryTextReversed": "--color-intent-secondary-text-reversed",
    "successAction": "--color-intent-success-action",
    "successBackground": "--color-intent-success-background",
    "successText": "--color-intent-success-text",
    "tertiaryText": "--color-intent-tertiary-text",
    "tertiaryTextReversed": "--color-intent-tertiary-text-reversed",
    "transparent": "--color-intent-transparent",
    "warningAction": "--color-intent-warning-action",
    "warningBackground": "--color-intent-warning-background",
    "warningText": "--color-intent-warning-text",
  },
  "colors": Object {
    "accent1": Object {
      "100": "--color-accent1-100",
      "300": "--color-accent1-300",
      "500": "--color-accent1-500",
      "700": "--color-accent1-700",
      "900": "--color-accent1-900",
      "baseHue": "--color-accent1-base-hue",
      "baseSat": "--color-accent1-base-sat",
    },
    "accent2": Object {
      "100": "--color-accent2-100",
      "300": "--color-accent2-300",
      "500": "--color-accent2-500",
      "700": "--color-accent2-700",
      "900": "--color-accent2-900",
      "baseHue": "--color-accent2-base-hue",
      "baseSat": "--color-accent2-base-sat",
    },
    "accent3": Object {
      "100": "--color-accent3-100",
      "300": "--color-accent3-300",
      "500": "--color-accent3-500",
      "700": "--color-accent3-700",
      "900": "--color-accent3-900",
      "baseHue": "--color-accent3-base-hue",
      "baseSat": "--color-accent3-base-sat",
    },
    "current": "--color-current",
    "effectTransparentHeavy": "--color-effect-transparent-heavy",
    "effectTransparentLight": "--color-effect-transparent-light",
    "effectTransparentMedium": "--color-effect-transparent-medium",
    "grey": Object {
      "0": "--color-grey-0",
      "0-reversed": "--color-grey-0-reversed",
      "100": "--color-grey-100",
      "100-reversed": "--color-grey-100-reversed",
      "1000": "--color-grey-1000",
      "200": "--color-grey-200",
      "200-reversed": "--color-grey-200-reversed",
      "25": "--color-grey-25",
      "300": "--color-grey-300",
      "400": "--color-grey-400",
      "50": "--color-grey-50",
      "500": "--color-grey-500",
      "600": "--color-grey-600",
      "700": "--color-grey-700",
      "800": "--color-grey-800",
      "900": "--color-grey-900",
      "baseHue": "--color-grey-base-hue",
      "baseSat": "--color-grey-base-sat",
    },
    "primary": Object {
      "100": "--color-primary-100",
      "300": "--color-primary-300",
      "500": "--color-primary-500",
      "700": "--color-primary-700",
      "900": "--color-primary-900",
      "baseHue": "--color-primary-base-hue",
      "baseSat": "--color-primary-base-sat",
    },
    "transparent": "--color-transparent",
    "utilityGreen": Object {
      "100": "--color-utility-green-100",
      "300": "--color-utility-green-300",
      "500": "--color-utility-green-500",
      "700": "--color-utility-green-700",
      "900": "--color-utility-green-900",
      "baseHue": "--color-utility-green-base-hue",
      "baseSat": "--color-utility-green-base-sat",
    },
    "utilityRed": Object {
      "100": "--color-utility-red-100",
      "300": "--color-utility-red-300",
      "500": "--color-utility-red-500",
      "700": "--color-utility-red-700",
      "900": "--color-utility-red-900",
      "baseHue": "--color-utility-red-base-hue",
      "baseSat": "--color-utility-red-base-sat",
    },
    "utilityYellow": Object {
      "100": "--color-utility-yellow-100",
      "300": "--color-utility-yellow-300",
      "500": "--color-utility-yellow-500",
      "700": "--color-utility-yellow-700",
      "900": "--color-utility-yellow-900",
      "baseHue": "--color-utility-yellow-base-hue",
      "baseSat": "--color-utility-yellow-base-sat",
    },
  },
}
`;
